FlowCal/plot.py

```diff
@@ -64,6 +67,510 @@
+    # attribute.
+    base = 10
+
+    def __init__(self, T=None, M=None, W=None, data=None, channel=None):
+        matplotlib.transforms.Transform.__init__(self)
+        # If data is included, try to obtain T, M and W from it
+        if data is not None:
+            if channel is None:
+                ValueError("if data is provided, a channel should be specified")
+            # Convert to list if necessary
+            if not isinstance(data, list):
+                data = [data]
+            # Obtain T, M, and W if not specified
+            # If elements of data have ``.range()``, use it to determine the
+            # max data value. Else, use the maximum value in the array.
+            if T is None:
+                T = 0
+                for d in data:
+                    # Extract channel
+                    y = d[:, channel] if d.ndim > 1 else d
+                    if hasattr(y, 'range') and hasattr(y.range, '__call__'):
+                        Ti = y.range(0)[1]
+                    else:
+                        Ti = np.max(y)
+                    T = Ti if Ti > T else T
+            if M is None:
+                M = 4.5 / np.log10(262144) * np.log10(T)
+            if W is None:
+                W = 0
+                for d in data:
+                    # Extract channel
+                    y = d[:, channel] if d.ndim > 1 else d
+                    # If negative events are present, use minimum.
+                    if np.any(y < 0):
+                        r = np.min(y)
+                        Wi = (M - np.log10(T / abs(r))) / 2
+                        W = Wi if Wi > W else W
+        else:
+            # Default parameter values
+            if T is None:
+                T = 262144
+            if M is None:
+                M = 4.5
+            if W is None:
+                W = 0.5
+        # Check that property values are valid
+        if T <= 0:
+            ValueError("T should be positive")
+        if M <= 0:
+            ValueError("M should be positive")
+        if W < 0:
+            ValueError("W should not be negative")
+
+        # Store parameters
+        self._T = T
+        self._M = M
+        self._W = W
+
+        # Calculate dependent parameter p
+        # It is not possible to analytically obtain ``p`` as a function of W
+        # only, so ``p`` is calculated numerically using a root finding
+        # algorithm. The initial estimate provided to the algorithm is taken
+        # from the asymptotic behavior of the equation as ``p -> inf``. This
+        # results in ``W = 2*log10(p)``.
+        p0 = 10**(W / 2.)
+        # Functions to provide to the root finding algorithm
+        def W_f(p):
+            return 2*p / (p + 1) * np.log10(p)
+        def W_root(p, W_target):
+            return W_f(p) - W_target
+        # Find solution
+        sol = scipy.optimize.root(W_root, x0=p0, args=(W))
+        # Solution should be unique
+        assert sol.success
+        assert len(sol.x) == 1
```

Comment:
Is there a reason to use `assert`s here? (as per discussion in #79, raising errors is preferred)